### PR TITLE
Add checkpoint feature

### PR DESCRIPTION
**Pull request by tjwatson**
This is a work in progress to merge the criu branch into integration.

### CODE REVIEW

File: dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops [hunk 1]
```diff
@@ -708,6 +708,10 @@ warning.earlyRelease=CWWKE0953W: This version of {0} is an unsupported early rel
 warning.earlyRelease.explanation=New features that are not fully tested are made available in early releases to elicit feedback before they are included in a supported release.
 warning.earlyRelease.useraction=Do not use an early release version on production systems, since early release versions are not supported.
 
+error.invalidPhaseName=CWWKE0954E: An empty or unknown checkpoint phase "{0}" was specified.
+error.invalidPhaseName.explanation=An unknown phase was passed in as the value of the checkpoint option.
```
Comment (jantley-ibm) [inline dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops]: change
passed in
to
specified

File: dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#COMPONENTPREFIX CWWKC                 
+#COMPONENTNAMEFOR CWWKC Checkpoint
+
+# Message numbers CWWKC0451-550 are reserved for this file.
+# NLS_MESSAGEFORMAT_VAR                
+# MSG_DESCRIPTIVE_NAME_CWWKC0000=CWWKC0000I: This is a message with inserts {0}
+# MSG_DESCRIPTIVE_NAME_CWWKC0000.explanation=Explanation text for the message
+# MSG_DESCRIPTIVE_NAME_CWWKC0000.useraction=User action text for the message
+
+CHECKPOINT_DUMP_INITIATED_CWWKC0451=CWWKC0451I: A server checkpoint was requested. The server will stopped when the checkpoint completes.
```
Comment (jantley-ibm) [inline dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops]: change
The server will stopped
to
The server will stop

File: dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+CHECKPOINT_DUMP_INITIATED_CWWKC0451.explanation=A checkpoint was requested. The server is halted and its current state is written to disk.
+CHECKPOINT_DUMP_INITIATED_CWWKC0451.useraction=Check the knowledge center for details of the checkpoint/restore operations.
```
Comment (jantley-ibm) [inline dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops]: change
knowledge center
to
documentation

File: dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+#
+CHECKPOINT_RESTORE_CWWKC0452I=CWWKC0452I: The liberty server is resuming execution from a checkpoint.
```
Comment (jantley-ibm) [inline dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops]: change
liberty server is resuming execution
to
Liberty server is resuming operation

File: dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+CHECKPOINT_RESTORE_CWWKC0452I.explanation=A restore operation was run on a liberty server checkpoint image.
```
Comment (jantley-ibm) [inline dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops:29]: change
liberty
to
Liberty

File: dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+CHECKPOINT_RESTORE_CWWKC0452I.useraction=Check the knowledge center for details of the checkpoint/restore operations.
```
Comment (jantley-ibm) [inline dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops]: change
knowledge center
to
documentation

File: dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops [hunk 7]
```diff
@@ -231,6 +231,11 @@ UPDATE_INDIRECT_CONFLICT_INCOMPATIBLE_FEATURES_SAME_PLATFORM_ERROR=CWWKF0047E: T
 UPDATE_INDIRECT_CONFLICT_INCOMPATIBLE_FEATURES_SAME_PLATFORM_ERROR.explanation=The server configuration contains an incompatible combination of features for the indicated programming models. This conflict can happen if the two incompatible features are configured directly in the server configuration, or if the configured features include different versions of a feature that are incompatible.
 UPDATE_INDIRECT_CONFLICT_INCOMPATIBLE_FEATURES_SAME_PLATFORM_ERROR.useraction=Try to specify a compatible version of the configured features that caused the conflict in the server.xml file. If no other versions are available, then the configured features are not compatible with each other and cannot be configured at the same time.
 
+#CWWKF0048E
+CHECKPOINT_REQUESTED_CHECKPOINT_FEATURE_MISSING=CWWKF0048E: The server command specified the checkpoint action. A checkpoint cannot be taken because the checkpoint-1.0 feature was not configured in the server.xml file.
```
Comment (jantley-ibm) [inline dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops]: change
was not
to
is not

File: dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops [hunk 8]
```diff
@@ -231,6 +231,11 @@ UPDATE_INDIRECT_CONFLICT_INCOMPATIBLE_FEATURES_SAME_PLATFORM_ERROR=CWWKF0047E: T
...
+CHECKPOINT_REQUESTED_CHECKPOINT_FEATURE_MISSING.explanation=The server checkpoint action requires that the checkpoint-1.0 feature be configured.
```
Comment (jantley-ibm) [inline dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops:236]: Change to
The checkpoint-1.0 feature must be configured to use a server checkpoint action.

**Review by jantley-ibm — COMMENTED** [review]
ID edits provided, let me know if you have questions. I will add the "ID reviewed" tag when complete.

File: dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops (hunk 1 above)
Comment (tjwatson) [reply dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops]: Should be fixed now.

File: dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops (hunk 3 above)
Comment (tjwatson) [reply dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops]: Fixed.

File: dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops (hunk 5 above)
Comment (tjwatson) [reply dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops:29]: Fixed.

File: dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops (hunk 4 above)
Comment (tjwatson) [reply dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops]: Fixed.

File: dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops (hunk 2 above)
Comment (tjwatson) [reply dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops]: Fixed.

File: dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops (hunk 6 above)
Comment (tjwatson) [reply dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops]: Fixed.

File: dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops (hunk 7 above)
Comment (tjwatson) [reply dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops]: Fixed.

File: dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops (hunk 8 above)
Comment (tjwatson) [reply dev/com.ibm.ws.kernel.feature.core/resources/com/ibm/ws/kernel/feature/internal/resources/ProvisionerMessages.nlsprops:236]: Fixed

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
@dmuelle thanks for doing the review of the initial messages for checkpoint.  I think I have addressed your comments.  In addition we have added more messages to `dev/io.openliberty.checkpoint/resources/io/openliberty/checkpoint/resources/CheckpointMessages.nlsprops`.  Please have a fresh look, thanks.

**Comment by skasund** [comment]
L2 message review not required.

**Comment by tjwatson** [comment]
Closing this in favor of the rebase/squash approach in PR #19856

**Pull request closed without merging**